packages/office-addin-unit-test/src/officeJSMock.ts

```diff
@@ -0,0 +1,150 @@
+import { usageDataObject } from "./defaults";
+
+export class OfficeJSMock {
+  constructor(object?: OfficeObject) {
+    this.properties = new Map<string, OfficeJSMock>();
+    this.loaded = false;
+    this.resetValue(undefined);
+    if (object) {
+      this.populate(object);
+    }
+  }
+
+  // Adds a function to OfficeJSMock
+  addMockFunction(methodName: string, functionality?: Function) {
+    this[methodName] = functionality ? functionality : function () {};
+  }
+
+  // Adds a OfficeJSMock to OfficeJSMock
+  addMockObject(objectName: string) {
+    const officeJSMock = new OfficeJSMock();
+    officeJSMock.isMockObject = true;
+    this.properties.set(objectName, officeJSMock);
+    this[objectName] = this.properties.get(objectName);
+  }
+
+  // load of Office.js API
+  load(property: string) {
+    this.loadMultipleProperties(property);
+  }
+
+  // Adds a property of any type to OfficeJSMock
+  setMock(propertyName: string, value: unknown) {
+    if (!this.properties.has(propertyName)) {
+      const officeJSMock = new OfficeJSMock();
+      officeJSMock.isMockObject = false;
+      this.properties.set(propertyName, officeJSMock);
+    }
+    this.properties.get(propertyName)?.resetValue(value);
+    this[propertyName] = this.properties.get(propertyName)?.value;
+  }
+
+  // sync of Office.js API
+  sync() {
+    this.properties.forEach((property: OfficeJSMock, key: string) => {
+      property.sync();
+      this.assignValue(key);
+    });
+    if (this.loaded) {
+      this.value = this.valueBeforeLoaded;
+    }
+  }
+
+  private assignValue(propertyName: string) {
+    if (this.properties.get(propertyName)?.isMockObject) {
+      this[propertyName] = this.properties.get(propertyName);
+    } else {
+      this[propertyName] = this.properties.get(propertyName)?.value;
+    }
+  }
+
+  private loadCalled() {
+    this.loaded = true;
+    this.value = `Error, context.sync() was not called`;
+  }
+
+  private loadMultipleProperties(property: string) {
+    property
+      .replace(/\s/g, "")
+      .split(",")
+      .forEach((individualProperties: string) => {
+        this.loadNavigational(individualProperties);
+      });
```

Comment:
Is it properties or navigational?  I don't see the connection.  

In general it looks like your using the term property and properties in a mixed way and I'm uncertain at a given point in the code what it's referring to.

Reply:
Yeah, I do agree, it is kinda hard to name them all in a meaningful way. I changed some names to see if it gets a little easier to understand.

The problem is that we have the following structure:
- Complete properties, separated each by commas (note that completely property can be navigational or not)
- Complete property: navigational, which contains several properties and one scalar, or just a scalar.